CS483/CS483/Kartaclysm/StateMachine/Gameplay/StatePlayerSelectionMenu.cpp

```diff
@@ -138,6 +131,30 @@ void Kartaclysm::StatePlayerSelectionMenu::Update(const float p_fDelta)
 				m_mPerPlayerMenuState[i].pReadyButton = m_pGameObjectManager->CreateGameObject("CS483/CS483/Kartaclysm/Data/Menus/PlayerSelectionMenu/player_" + strPlayerNum + "/not_ready_" + strPlayerNum + ".xml");
 			}
 		}
+		else if (bCancel)
+		{
+			if (m_mPerPlayerMenuState[i].bJoined)
+			{
+				RemovePlayer(i);
+
+				// Pop if no player is left joined. Otherwise, reset all players to not ready.
+				bool bPlayerStillJoined = false;
+				for (auto mPlayer : m_mPerPlayerMenuState)
+				{
+					if (mPlayer.bJoined)
+					{
+						bPlayerStillJoined = true;
+						mPlayer.bReady = false;
```

Comment:
If you're resetting all players to not being ready, you also need to change the "Ready" icons from green to red.